src/puppetlabs/puppetdb/constants.clj

```diff
@@ -0,0 +1,4 @@
+(ns puppetlabs.puppetdb.constants)
+
+(def filename-forbidden-characters
```

Comment:
OK, so at first I was a little wary of this, since it's not the unicode characters we're worried about, it's the possibility that the corresponding "ascii" bytes might show up in the utf-8 encoding of some unicode string, but maybe it's actually fine given the way utf-8 works (thinking out loud...), i.e. because the 7-bit ascii range "passes-through", covering "/" and ":", and the only null byte in standard utf-8 is for unicode null itself?  If that's right, then this approach is perhaps OK for anything in the 7-bit range, including null, but might or might not be for any other chars.

(Also had no idea bout MUTF-8 https://en.wikipedia.org/wiki/UTF-8#Modified_UTF-8)